agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/logging/WarningLogger.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.opentelemetry.exporter.implementation.logging;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.util.annotation.Nullable;
+
+public class WarningLogger {
+
+  private final Logger logger;
+  private final String message;
+  private final AtomicLong numWarnings = new AtomicLong();
+
+  // Period for scheduled executor
+  private final int intervalSeconds;
+
+  private final AtomicBoolean firstFailure = new AtomicBoolean();
+
+  public WarningLogger(Class<?> source, String message) {
+    this(source, message, 300);
+  }
+
+  // visible for testing
+  WarningLogger(Class<?> source, String message, int intervalSeconds) {
+    logger = LoggerFactory.getLogger(source);
+    this.message = message;
+    this.intervalSeconds = intervalSeconds;
+  }
+
+  public void recordWarning() {
```

Comment:
pass in the message body instead so that it can be reused over and over again.

Reply:
linked to this comment also in the work item above